src/Workspaces/Core/Portable/Workspace/Host/DocumentService/IDocumentExcerptService.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    /// <summary>
+    /// excerpt some part of <see cref="Document"/>
+    /// </summary>
+    internal interface IDocumentExcerptService : IDocumentService
+    {
```

Comment:
what is an IDocumentService?

Reply:
IWorkspaceService is a service that can have different behavior (implementation) per workspace
ILagnuageService is a service that can have different behavior (implementation) per language/project
IDocumentService is a service that can have different behavior (implementation) per roslyn document.